File: apzkr-pzpi-21-4-bondarenko-taras/Task4-MobileClient/CargoTruckMobileClient/app/src/main/java/com/example/cargotruckmobile/Network/StatisticsService.kt
package com.example.cargotruckmobile.Network

import com.example.cargotruckmobile.Models.Container
import com.example.cargotruckmobile.Models.TripData
import retrofit2.http.GET
import retrofit2.http.Path

interface StatisticsService {

    @GET("statistics/delivery/container/{containerId}")
    suspend fun getContainerStatistics(@Path("containerId") containerId: String): Container

    @GET("statistics/trip/{tripId}")
    suspend fun getTripStatistics(@Path("tripId") tripId: String): TripData
}

File: apzkr-pzpi-21-4-bondarenko-taras/Task3-WebClient/CargoTruckWebClient/src/Api/StatisticsService.js

import { ApiClient } from './ApiClient';

export const StatisticsService = {

    getContainerStatistics: async (id) => {
    const response = await ApiClient.get(`statistics/delivery/container/${id}`);
    return response.data;
  },
  getTripStatistics: async (id) => {
    const response = await ApiClient.get(`statistics/trip/${id}`);
    return response.data;
  }
};
